utxo/src/utxo_impl/utxo_storage/view_impls.rs

```diff
@@ -0,0 +1,124 @@
+use common::{
+    chain::{GenBlock, OutPoint},
+    primitives::Id,
+};
+
+use crate::{FlushableUtxoView, Utxo, UtxosCache, UtxosView};
+
+use super::{UtxosDB, UtxosDBMut, UtxosStorageRead, UtxosStorageWrite};
+
+mod utxosdb_utxosview_impls {
+    use common::{
+        chain::{GenBlock, OutPoint},
+        primitives::Id,
+    };
+
+    use crate::{utxo_storage::UtxosStorageRead, Utxo, UtxosCache, UtxosView};
+
+    use super::*;
+    pub fn utxo<S: UtxosStorageRead>(db: &S, outpoint: &OutPoint) -> Option<Utxo> {
+        match db.get_utxo(outpoint) {
+            Ok(res) => res,
+            Err(e) => {
+                panic!(
+                    "Database error while attempting to retrieve utxo from the database: {}",
+                    e
+                );
+            }
+        }
```

Comment:
Not sure how much better it is but I think the pattern here could be replaced with something like;
```rust
db.get_utxo(outpoint).unwrap_or_else(|e| panic!(...))
```